resources/java.rb

```diff
@@ -66,7 +66,7 @@
     if new_resource.install_directory != def_install
       options_array << 'INSTALLDIR=' + install_directory
     end
-    if !new_resource.proxy_hostname.empty?
+    if !new_resource.proxy_hostname.to_s.empty?
```

Comment:
I think this should be `if !new_resource.proxy_hostname.nil?` or just `if new_resource.proxy_hostname`

Reply:
Not sure if .nil? will capture the empty string case.  I do think just a truthy test like `if new_resource.proxy_name` might work.  A this point I've tested the code above.

Reply:
I would prefer the simple test. This seems a little convoluted, and I don't see any other tests in the cookbook that look like this.